### PR TITLE
support fetchr stats collection and monitoring

### PR DESCRIPTION
**Pull request by lingyan**
This allows apps to provide their custom function to collect stats for fetchr requests.

@yahoo/team-fluxible

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: libs/fetcher.js [hunk 1]
```diff
@@ -130,33 +135,59 @@ Request.prototype.clientConfig = function (config) {
...
+            resource: self.resource,
+            operation: self.operation,
+            params: self._params,
+            statusCode: err ? err.statusCode : (result && result.meta && result.meta.statusCode || 200),
+            err: err,
+            time: Date.now() - self._startTime
+        };
+        statsCollector(stats);
+    }
+};
```
Comment (lingyan) [inline libs/fetcher.js]: `err` and `result` objects do not have the same structure on client and server.  So had to duplicate this function here to keep code simple.  Looks like `Request` classes on server and client share a lot of common things.  We should refactor `fetchr.js` and `fetchr.client.js` to pull out common code and normalize data structure when we get time.

Comment (redonkulus) [inline libs/fetcher.js]: I thought the `err` and `result` objects were recently aligned with BC to ensure we didn't break anyone. But I agree the two files needs to be merged, its too easy to break now.

**Comment by redonkulus** [comment]
Code looks good to me 👍

File: libs/fetcher.js (hunk 1 above)
Comment (lingyan) [inline libs/fetcher.js]: We aligned `err` and `result` for the callback of `Request.prototype.end()`, not the `err` and `result` from `executeRequest` (private method for executing request).

**Pull request merged**